CRADLE/CorrectBias/calculateOnebp.pyx

```diff
@@ -1337,20 +1318,17 @@ cpdef correctReadCount(covariFileName, chromo, analysisStart, analysisEnd, lastB
 		rcArr = rcArr[selectedIdx]
 
 		idx = np.where( (rcArr < np.finfo(np.float32).min) | (rcArr > np.finfo(np.float32).max))
-		if len(idx[0]) > 0:
-			tempStarts = np.delete(starts, idx)
-			rcArr = np.delete(rcArr, idx)
-			if len(rcArr) > 0:
-				writeBedFile(subfinalCtrlNames[rep], tempStarts, rcArr, analysisEnd)
-			else:
-				os.remove(subfinalCtrlNames[rep])
-				subfinalCtrlNames[rep] = None
-		else:
-			if len(rcArr) > 0:
-				writeBedFile(subfinalCtrlNames[rep], starts, rcArr, analysisEnd)
-			else:
-				os.remove(subfinalCtrlNames[rep])
-				subfinalCtrlNames[rep] = None
+		if len(rcArr) > 0:
+			with tempfile.NamedTemporaryFile(mode="w+t", suffix=".bed", dir=vari.OUTPUT_DIR, delete=False) as subfinalCtrlFile:
+				subfinalCtrlNames[rep] = subfinalCtrlFile.name
+
+				if len(idx[0]) > 0:
+					tempStarts = np.delete(starts, idx)
+					rcArr = np.delete(rcArr, idx)
+				else:
+					tempStarts = starts
+
```

Comment:
is this okay? 

```python
tempStarts = np.delete(starts, idx)
rcArr = np.delete(rcArr, idx)
if len(rcArr) > 0:
    with tempfile.NamedTemporaryFile(mode="w+t", suffix=".bed", dir=outputDir, delete=False) as subfinalCtrlFile:
        subfinalCtrlNames[rep] = subfinalCtrlFile.name
        writeBedFile(subfinalCtrlFile, tempStarts, rcArr, analysisEnd, binsize)
```

I don't think you need the `idx[0] > 0` check, since
```python
tempStarts = np.delete(starts, idx)
rcArr = np.delete(rcArr, idx)
```
just won't do anything (tempStarts will equal starts; rcArr won't change) in that case.

Reply:
that works!

Reply:
Could you do the same thing in the code velow as well?

Reply:
Yeah. I need to change this in CorrectBiases_stored, too.